### PR TITLE
IG-19426: fix client.write() when first column is all null

### PR DESCRIPTION
**Pull request by gtopper**

### CODE REVIEW

File: column.go [hunk 1]
```diff
@@ -57,6 +57,8 @@ func (c *colImpl) Len() int {
 		return len(c.msg.Times)
 	case pb.DType_BOOLEAN:
 		return len(c.msg.Bools)
+	case pb.DType_NULL:
+		return -1
```
Comment (dinal) [inline column.go:61]: it already returned -1 (out of the switch case)

File: frame.go [hunk 2]
```diff
@@ -173,11 +173,7 @@ func (fr *frameImpl) Labels() map[string]interface{} {
 
 // Len is the number of rows
 func (fr *frameImpl) Len() int {
-	if len(fr.columns) > 0 {
-		return fr.columns[0].Len()
-	}
-
-	return 0
+	return fr.indices[0].Len()
```
Comment (dinal) [inline frame.go:176]: do we always have something in indices that you can access the array without checking it's length?
what about empty df?

**Comment by dinal** [comment]
add a test to cover the bug

File: column.go (hunk 1 above)
Comment (gtopper) [reply column.go:61]: I know. I wanted to make it explicit and not a "panic? 🤷‍♂️" thing.

File: frame.go (hunk 2 above)
Comment (gtopper) [reply frame.go:176]: Every dataframe has an index.

Comment (gtopper) [reply frame.go:176]: Ok, turns out this can happen not because of pandas but because of how we populate the structure internally...

**Review by dinal — APPROVED** [review]

**Pull request merged**